### PR TITLE
Fix #17185: MusicXML ignores custom string data

### LINKED ISSUE #17185: MusicXML import ignores custom string data
### Issue type

Other type of issue

### Bug description

I have changed the string data under the Staff/Part properties dialogue for my solo guitar part on Musescore 4. Unfortunatlely this data isn't saved when exporting to MusicXML. If I open this dialogue on the imported MusicXML file, the string data is back to standard tuning.

### Steps to reproduce

1. Create a Guitar with a linked 6 line tablature staff in Musescore 4.
2. Open the staff/part dialogue, click Edit String Data... and change one of the strings to a different note.
3. Export the file to musicXML
4. Open the musicXML file with musescore 4.
5. Click the Edit String Data... dialogue box. The strings are back to standard tuning (E A D G B E)

### Screenshots/Screen recordings

_No response_

### MuseScore Version

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230651553, revision: github-musescore-musescore-dbe7c6d

### Regression

I don't know

### Operating system

Windows 11

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HemantAntony**
Resolves: #17185 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by HemantAntony** [comment]
This PR also solves another unreported problem. When you tune a string to a -1 octave, say C-1 and export it to MusicXML, the exported file gets corrupted because the `minInclusive` value of `tuning-octave` in the schema is 0

**Comment by Jojo-Schmitz** [comment]
I don't quite understand how the changes of this PR could lead to those unit test failures, which are about the number of frets but not their tuning? But I guess just adjusting that test is the correct way of dealing with this

**Comment by lvinken** [comment]
A quick check shows at least the staff tuning is imported correctly with this PR.

Please also add one or more test files.

**Comment by DmitryArefiev** [comment]
#17185 also looks fine on my side (Windows10) , but tests should be fixed at first. @HemantAntony Can you look into it please?

**Comment by Jojo-Schmitz** [comment]
That mtest failure is related to the 1st commit, not to the 2nd

**Comment by HemantAntony** [comment]
Fixed the tests and updated the PR. I also added two tests to test both problems (#17185 and negative octave).

Also should I add PDFs of the test files? I see there are PDFs of `xml` and `mscx`

**Review by RomanPudashkin — APPROVED** [review]

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -6775,6 +6775,19 @@ void ExportMusicXml::writeElement(EngravingItem* el, const Measure* m, staff_idx
...
+//  clampMusicXmlOctave
+//---------------------------------------------------------
+
+/**
+ Clamps octave to min and max value as per MusicXML Schema
+ */
+
+static void clampMusicXmlOctave(int& octave)
+{
+    octave = std::clamp(octave, 0, 9);
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:6788]: This seems to be C++17 and later only, am I right that `octave = std::max(std::min(octave, 9), 0);` is the same thing for older C++ standards?

**Comment by DmitryArefiev** [comment]
Tested #17185 on Win10, Mac13 - **FIXED**    

@HemantAntony Thanks for fixing!
